examples/RNOneSignal/App.js

```diff
@@ -66,80 +66,86 @@ export default class App extends Component {
             isEmailLoading: false,
 
             // In-App Messaging states
-            iam_paused: false,
-
-            // Add more states here...
+            iamPaused: false,
 
             // Demo App states
             debugText: ''
+
+            // Add more states here...
         };
 
-        // Log level logcat is 6 (VERBOSE) and log level alert is 0 (NONE)
-        OneSignal.setLogLevel(6, 0);
+    }
 
-        // Share location of device
-        OneSignal.setLocationShared(true);
+    /* R E A C T  L I F E C Y C L E */
 
+    async componentDidMount() {
+        console.log("Mounted!");
+        /* O N E S I G N A L  S E T U P */
+        // Log level logcat is 6 (VERBOSE) and log level alert is 0 (NONE)
+        OneSignal.setLogLevel(6, 0);
+        OneSignal.setAppId(appId);
         OneSignal.setRequiresUserPrivacyConsent(requiresPrivacyConsent);
-        OneSignal.init(appId, {
-            kOSSettingsKeyAutoPrompt: true,
+        OneSignal.setLocationShared(true);
+        OneSignal.promptForPushNotificationsWithUserResponse(response => {
+            console.log("Prompt response:", response);
+        });
+        /* O N E S I G N A L  H A N D L E R S */
+        OneSignal.setNotificationWillShowInForegroundHandler(notifReceivedEvent => {
+            console.log("OneSignal: notification will show in foreground:", notifReceivedEvent);
+            let notif = notifReceivedEvent.getNotification();
+            // future: modify the notification
```

Comment:
let's omit this from the example for now, since this isn't planned immediately after the major release. There's a good chunk of work on the native SDK side that is needed to support this.

Reply:
Done